### PR TITLE
Introduce df.spark.coalesce()

### PR DESCRIPTION
**Pull request by xinrong-databricks**
Introduce df.spark.coalesce()

### CODE REVIEW

File: databricks/koalas/spark/accessors.py [hunk 1]
```diff
@@ -939,11 +939,38 @@ def repartition(self, num_partitions: int) -> "ks.DataFrame":
...
+
+        .. note:: This operation results in a narrow dependency, e.g. if you go from 1000
+        partitions to 100 partitions, there will not be a shuffle, instead each of the 100 new
+        partitions will claim 10 of the current partitions. If a larger number of partitions is
+        requested, it will stay at the current number of partitions. However, if you're doing a
+        drastic coalesce, e.g. to num_partitions = 1, this may result in your computation taking
+        place on fewer nodes than you like (e.g. one node in the case of num_partitions = 1). To
+        avoid this, you can call repartition(). This will add a shuffle step, but means the
+        current upstream partitions will be executed in parallel (per whatever the current
+        partitioning is).
```
Comment (HyukjinKwon) [inline databricks/koalas/spark/accessors.py:957]: I believe you have to add four spaces for the lines after `.. note::`.

Comment (xinrong-databricks) [reply databricks/koalas/spark/accessors.py]: Thank you! Updated.

File: databricks/koalas/spark/accessors.py [hunk 2]
```diff
@@ -939,11 +939,38 @@ def repartition(self, num_partitions: int) -> "ks.DataFrame":
...
+            partitioning is).
+
+        Parameters
+        ----------
+        num_partitions : int
+            The target number of partitions.
+
+        Returns
+        -------
+        DataFrame
```
Comment (itholic) [inline databricks/koalas/spark/accessors.py:966]: Maybe could we have some examples ??

File: databricks/koalas/tests/test_frame_spark.py [hunk 3]
```diff
@@ -92,3 +92,32 @@ def test_repartition(self):
         new_kdf = kdf.spark.repartition(num_partitions)
         self.assertEqual(new_kdf.to_spark().rdd.getNumPartitions(), num_partitions)
         self.assert_eq(kdf.sort_index(), new_kdf.sort_index())
+
+    def test_coalesce(self):
+        num_partitions = 10
+        kdf = ks.DataFrame({"age": [5, 5, 2, 2], "name": ["Bob", "Bob", "Alice", "Alice"]})
+        kdf = kdf.spark.repartition(num_partitions)
+
+        new_kdf = kdf.spark.coalesce(--num_partitions)
```
Comment (itholic) [inline databricks/koalas/tests/test_frame_spark.py:101]: Hmm... I don't think maybe Python doesn't support such decrement operator ??

```python
>>> num_partitions = 10
>>> --num_partitions
10
```

Maybe I think you should specify '9' if you want give '9' as a parameter.

**Review by itholic — APPROVED** [review]
Otherwise, seems good to me

File: databricks/koalas/tests/test_frame_spark.py (hunk 3 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_frame_spark.py]: Thanks for catching this! Modified.

**Pull request merged**